ecoscope/io/earthranger.py

```diff
@@ -922,3 +798,94 @@ def patch_event(
         else:
             results = [self._patch(f"activity/event/{event_id}", payload=events)]
         return pd.DataFrame(results)
+
+    def post_subject(
+        self,
+        subject_name: str,
+        subject_type: str,
+        subject_subtype: str,
+        is_active: bool = True,
+        tracks_available: bool = True,
+        **kwargs,
+    ) -> pd.DataFrame:
+        """
+        Parameters
+        ----------
+        subject_name
+        subject_type
+        subject_subtype
+        is_active
+        tracks_available
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+
+        payload = {
+            "name": subject_name,
+            "subject_type": subject_type,
+            "subject_subtype": subject_subtype,
+            "is_active": is_active,
+            "tracks_available": tracks_available,
+        }
+
+        if kwargs:
+            payload.update(kwargs)
+
+        response = self._post("subjects", payload=payload)
+        return pd.DataFrame([response])
+
+    def post_source(
+        self,
+        source_type: str,
+        manufacturer_id: str,
+        model_name: str,
+        provider: str,
+        additional: typing.Dict = {},
+        **kwargs,
+    ) -> pd.DataFrame:
+        """
+        Parameters
+        ----------
+        source_type
+        manufacturer_id
+        model_name
+        provider
+        additional
+        Returns
+        -------
+        pd.DataFrame
+        """
+
+        payload = {
+            "source_type": source_type,
+            "manufacturer_id": manufacturer_id,
+            "model_name": model_name,
+            "additional": additional,
+            "provider": provider,
+        }
+
+        if kwargs:
+            payload.update(kwargs)
+
+        response = self._post("sources", payload=payload)
+        return pd.DataFrame([response])
+
+    def post_patrol(self, priority: int, **kwargs) -> pd.DataFrame:
```

Comment:
I see that patrols consist of patrol segments. Suppose we need a function for posting those as well? Are there any other parameters that would define a patrol, e.g., title etc that we can expose to the user?

Reply:
I added a post_patrol_segment function. Yes, there are also other parameters that can define a patrol (objective, title, files, notes). All of them can be specified through kwargs. I guess that title could be added as an optional parameter for the post_patrol function since it's a quite common patrol parameter?